workers/social/lib/social/models/group/index.coffee

```diff
@@ -1143,6 +1155,69 @@ module.exports = class JGroup extends Module
         @update { $set: dataToUpdate }, callback
 
 
+  setOAuth   : permit
+    advanced : [
+      { permission: 'edit own groups', validateWith : Validators.group.admin }
+      { permission: 'edit groups',     superadmin   : yes }
+    ]
+    success: (client, options, callback) ->
+
+      { enabled, provider, url, applicationId, applicationSecret } = options
+
+      OAuth = require '../oauth'
+      group = client?.context?.group
+
+      if not group or group is 'koding'
+        return callback new KodingError 'Session data invalid'
+
+      if not (_provider = OAuth.PROVIDERS[provider]) or not _provider.enabled
+        return callback new KodingError 'Provider not supported at this time.'
+
+      dataToUpdate = {}
+      dataToUpdate["config.#{provider}"] = {}
+
+      notifyOptions = { target: group, group }
+
+      if not enabled
+
+        @updateAndNotify notifyOptions, { $unset: dataToUpdate }, (err) =>
+          return callback err  if err
+
+          JForeignAuth = require '../foreignauth'
+          JForeignAuth.remove { group, provider }
+
+          @fetchData (err, data) ->
+            return callback err  if err
+
+            dataToUnset = {}
+            dataToUnset["data.#{provider}.applicationSecret"] = null
```

Comment:
i am not sure if it works with setting to null, but i am sure it works with empty string, if it is working ignore my comment